Sync `v2-3-stable` with `v2-3-test` to release `2.3.2`

ephraimbuddy opened the pull request:
Time for `2.3.2rc1`!

tirkarthi committed:
Add is_mapped field to Task response. (#23319)

* Add is_mapped field to Task response.

* Add is_mapped to schema file and add test for GetTasks.

(cherry picked from commit f3d80c2a0dce93b908d7c9de30c9cba673eb20d5)

jedcunningham committed:
Few fixes in the providers release doc (#23382)

(cherry picked from commit acbb09c7e10c1452629ff1ab7641a3fa72237338)

potiuk committed:
Fix cassandra to 3.0.25 (#23522)

fix cassandra to 3.0.25 as latest 3.0 (3.0.26) does not start cleanly

(cherry picked from commit 3cd37d4d206587314f96f0a3f7dfabaaef1b6cc6)

jedcunningham committed:
Change chart annotation generator to use RELEASE_NOTES (#23549)

(cherry picked from commit e2c7847c6bf73685f0576364787fab906397a6fe)

potiuk committed:
Temporarily pin xmltodict to 0.12.0 to fix main failure (#23577)

The xmltodict 0,13.0 breaks some tests and likely 0.13.0 is buggy
as the error is ValueError: Malformatted input.

We pin it to 0.12.0 to fix the main failing.

Related: #23576
(cherry picked from commit 975996157f4cc35490acc69d98ce6582d81af93a)

nsAstro committed:
Add exception to catch single line private keys (#23043)

(cherry picked from commit e63dbdc431c2fa973e9a4c0b48ec6230731c38d1)

potiuk committed:
Make provider doc preparation a bit more fun :) (#23629)

Previously you had to manually add versions when changelog was
modified. But why not to get a bit more fun and get the versions
bumped automatically based on your assesment when reviewing the
provideers rather than after looking at the generated changelog.

(cherry picked from commit cfa95af7e83b067787d8d6596caa3bc97f4b25bd)

dstandish committed:
Simplify flash message for _airflow_moved tables (#23635)

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit b68667843c7a9796e7017b729f4474c0b0c990cd)

potiuk committed:
Add tool to automaticaly update status of AIP-47 issues. (#23745)

(cherry picked from commit 91a96f164f5af7872d94fdeab8aed7c487976391)

potiuk committed:
Self upgrade when refreshing images (#23686)

When you have two branches, you should sefl-upgrade breeze to make
sure you use the version that is tied with your branch.

Usually we have two active branches - main and the last released
line, so switching between then is not unlikely for maintainers.

(cherry picked from commit e09e4635b0dc50cbd3a18f8be02ce9b2e2f3d742)

joelossher committed:
Exclude missing tasks from the gantt view (#23627)

* Exclude missing tasks from the gantt view

Stops the gantt view from crashing if a task no longer exists
in a DAG but there are TaskInstances for that task.

* Fix tests

(cherry picked from commit 4b731f440734b7a0da1bbc8595702aaa1110ad8d)

dstandish committed:
Mask sensitive values for not-yet-running TIs (#23807)

Alternative approach to #22754.  Resolves  #22738.

(cherry picked from commit 2dc806367c3dc27df5db4b955d151e789fbc78b0)

kaxil committed:
Doc: Add column names for DB Migration Reference (#23853)

Before the automation: https://airflow.apache.org/docs/apache-airflow/2.2.5/migrations-ref.html
Currently (with missing column names): https://airflow.apache.org/docs/apache-airflow/2.3.0/migrations-ref.html

(cherry picked from commit 344304e9e468446ee87c012eeae2ba0ec01acd47)

ashb committed:
Fix exception trying to display moved table warnings (#23837)

If you still have an old dangling table from the 2.2 migration this
would fail. Make it more resilient and cope with both styles of moved
table name

(cherry picked from commit 4829b2392586643f5f7393ed423afc28c4814b63)

potiuk committed:
Upgrade `pip` to 22.1.1 version (just released) (#23854)

(cherry picked from commit a7425dc62e89cb73ac2bf04bd706858c915a6a12)

potiuk committed:
Add better feedback to Breeze users about expected action timing (#23827)

There are a few actions in Breeze that might take more or less time
when invoked. This is mostly when you need to upgrade Breeze or
update to latest version of the image because some dependedncies
were added or image was modified.

While we have improved significantly the waiting time involved
now (and caching problems have been fixed to make it as fast
possible), there are still a few situations that you need to have
a good connectivity and a little time to run the upgrade. Which
is often not something you would like to loose your time on in
a number of cases when you need to do things fast.

Usually Breeeze does not force the user to perform such long
actions - it allows to continue without doing them (either by
timeout or by letting user answer "no" to question asked.

Previously Breeze have not informed the user about the exepcted
time of running such operation, but with this change it tells
what is the expected delay - thus allowing the user to make
informed action whether they want to run the upgrade or not.

(cherry picked from commit fe91db752525717cf5b8017f16e4bc60c6a2b26d)

kazanzhy committed:
Fix UnboundLocalError when sql is empty list in DbApiHook (#23816)

(cherry picked from commit 4b5a101d29d8b0188d41abc5d079b0d363902de1)

eskarimov committed:
Add Deferrable Databricks operators (#19736)

(cherry picked from commit 6150d283234b48f86362fd4da856e282dd91ebb4)

snjypl committed:
Add __wrapped__ property to _TaskDecorator (#23830)

Co-authored-by: Sanjay Pillai <sanjaypillai11 [at] gmail.com>
(cherry picked from commit a71e4b789006b8f36cd993731a9fb7d5792fccc2)

pierrejeambrun committed:
Highlight task states by hovering on legend row (#23678)

* Rework the legend row and add the hover effect.

* Move horevedTaskState to state and fix merge conflicts.

* Add tests.

* Order of item in the LegendRow, add no_status support

(cherry picked from commit 637a8b8af132e6231756160a3f6ce7ed789abaf6)

raphaelauv committed:
update K8S-KIND to 0.14.0 (#23859)

(cherry picked from commit 35620edd4b5b108adf355855e03224a08d132b10)

uranusjr committed:
Replaced all days_ago functions with datetime functions (#23237)

Co-authored-by: Dev232001 <thedevhooda@gmail.com>
(cherry picked from commit f352ee63a5d09546a7997ba8f2f8702a1ddb4af7)

alexkruc committed:
Ignore the DeprecationWarning in test_days_ago (#23875)

Co-authored-by: alexkru <alexkru@wix.com>
(cherry picked from commit eff697a5b89c74c995743f401f94d1171f39b08a)

potiuk committed:
Speed up Breeze experience on Mac OS (#23866)

This change should significantly speed up Breeze experience (and
especially iterating over a change in Breeze for MacOS users -
independently if you are using x86 or arm architecture.

The problem with MacOS with docker is particularly slow filesystem
used to map sources from Host to Docker VM. It is particularly bad
when there are multiple small files involved.

The improvement come from two areas:
* removing duplicate pycache cleaning
* moving MyPy cache to docker volume

When entering breeze we are - just in case - cleaning .pyc and
__pychache__ files potentially generated outside of the docker
container - this is particularly useful if you use local IDE
and you do not have bytecode generation disabled (we have it
disabled in Breeze). Generating python bytecode might lead to
various problems when you are switching branches and Python
versions, so for Breeze development where the files change
often anyway, disabling them and removing when they are found
is important. This happens at entering breeze and it might take
a second or two depending if you have locally generated.

It could happen that __init script was called twice (depending which
script was called - therefore the time could be double the one
that was actually needed. Also if you ever generated provider
packages, the time could be much longer, because node_modules
generated in provider sources were not excluded from searching
(and on MacOS it takes a LOT of time).

This also led to duplicate time of exit as the initialization code
installed traps that were also run twice. The traps however were
rather fast so had no negative influence on performance.

The change adds a guard so that initialization is only ever executed
once.

Second part of the change is moving the cache of mypy to a docker
volume rather than being used from local source folder (default
when complete sources are mounted). We were already using selective
mount to make sure MacOS filesystem slowness affects us in minimal
way - but with this change, the cache will be stored in docker
volume that does not suffer from the same problems as mounting
volumes from host. The Docker volume is preserved until the
`docker stop` command is run - which means that iterating over
a change should be WAY faster now - observed speed-up were around
5x speedups for MyPy pre-commit.

(cherry picked from commit ac9b224e629918c66e867cde8f94804b6336912c)

norm committed:
Enable clicking on DAG owner in autocomplete dropdown (#23804)

PR#18991 introduced directly navigating to a DAG when selecting one
from the typeahead search results. Unfortunately, the search results
also includes DAG owner names, and selecting one of those navigates to
a DAG with that name, which almost certainly doesn't exist.

This extends the autocompletion endpoint to return the type of result,
and adjusts the typeahead selection to use this to know which way to
navigate.

(cherry picked from commit 29afd35b9cfe141b668ce7ceccecdba60775a8ff)

potiuk committed:
Avoid printing exception when exiting tests command (#23897)

Fixes: #23868
(cherry picked from commit d788f4b90128533b1ac3a0622a8beb695b52e2c4)

potiuk committed:
Make --file command in static-checks autocomplete file name (#23896)

The --verbose and --dry-dun commands caused n --files command to fail
and the flag was "artifficial" -it was equivalent to bool flag.
the actual files were taken  from arguments.

This PR fixes it by turning the arguments into multiple ``--file``
commands  - each with its own completioin for local files.

(cherry picked from commit 1d53bec0618c042de5cd05801b0c1fe015e6b4f8)

igorborgest committed:
Fix Breeze documentation typo (#23919)

(cherry picked from commit 2e95153ec9c3cf54c4f2fc34f215f3ad3f65fc63)

jedcunningham committed:
Fix release date for 2.3.1

jedcunningham committed:
Make CI and PROD image builds consistent (#23841)

Simple refactoring to make the jobs more consistent.

(cherry picked from commit 30262b1fe3b095168c9b4fe5a78255048d2f410f)

ephraimbuddy committed:
Clarify manual merging of PR in release doc (#23928)

It was not clear to me what this really means

(cherry picked from commit bfe51ea64a6957587c33baf32916b205dea91a45)

eladkal committed:
Fix broken main (#23940)

main breaks with
`Traceback:
  /usr/local/lib/python3.7/importlib/__init__.py:127: in import_module
      return _bootstrap._gcd_import(name[level:], package, level)
  tests/providers/amazon/aws/hooks/test_cloud_formation.py:31: in <module>
      class TestCloudFormationHook(unittest.TestCase):
  tests/providers/amazon/aws/hooks/test_cloud_formation.py:67: in TestCloudFormationHook
      @mock_cloudformation
  /usr/local/lib/python3.7/site-packages/moto/__init__.py:30: in f
      module = importlib.import_module(module_name, "moto")
  /usr/local/lib/python3.7/importlib/__init__.py:127: in import_module
      return _bootstrap._gcd_import(name[level:], package, level)
  /usr/local/lib/python3.7/site-packages/moto/cloudformation/__init__.py:1: in <module>
      from .models import cloudformation_backends
  /usr/local/lib/python3.7/site-packages/moto/cloudformation/models.py:18: in <module>
      from .parsing import ResourceMap, OutputMap
  /usr/local/lib/python3.7/site-packages/moto/cloudformation/parsing.py:17: in <module>
      from moto.apigateway import models  # noqa  # pylint: disable=all
  /usr/local/lib/python3.7/site-packages/moto/apigateway/__init__.py:1: in <module>
      from .models import apigateway_backends
  /usr/local/lib/python3.7/site-packages/moto/apigateway/models.py:9: in <module>
      from openapi_spec_validator import validate_spec
  E   ModuleNotFoundError: No module named 'openapi_spec_validator'
  `
  Fix is already in placed in moto https://github.com/spulec/moto/pull/5165 but version 3.1.11 wasn't released yet

(cherry picked from commit 86d40a95c4b6085ddea7b4d14bb6d91c126138dc)

potiuk committed:
Add typing to Azure Cosmos Client Hook (#23941)

New release of Azure Cosmos library has added typing information
and it broke main builds with mypy verification.

(cherry picked from commit 3393647aa63cbfdd2e6b90b7a5c9971732a54fc2)

Taragolis committed:
Remove redundant register exit signals in `dag-processor` command (#23886)

(cherry picked from commit db890df38f3020dc2cd322d5d5f67f4e57e186fd)

potiuk committed:
Disable rebase workflow (#23943)

The change of the release workflow in #23928 removed the reason
why we should have rebase workflow possible. We only needed to
do rebase when we merged test branch into stable branch and
since we are doing it manually, there is no more reeason to
have it in the GitHub UI.

(cherry picked from commit 17488768bb84468d965cbce6b54186d6c76a1053)

bbovenzi committed:
Prevent UI from crashing if grid task instances are null (#23939)

* UI fix for null task instances

* improve tests without global vars

* fix test data

(cherry picked from commit 9314a4ab8bf11e56125cbca6b012d61d1002ac32)

jedcunningham committed:
Fix and speed up grid view (#23947)

This fetches all TIs for a given task across dag runs, leading to
signifincatly faster response times. It also fixes a bug where Nones
were being passed to the UI when a new task was added to a DAG with
exiting runs.

(cherry picked from commit 1cf483fa0c45e0110d99e37b4e45c72c6084aa97)

Commit:
Update dep for databricks #23917 (#23927)

(cherry picked from commit 80c3fcd097c02511463b2c4f586757af0e5f41b2)

mhenc committed:
Use '--subdir' argument value for standalong dag processor. (#23864)

(cherry picked from commit 188ec4a8d61bdb536c33c9aa518f5acc13b0896e)

potiuk committed:
Revert "Add limit for JPype1 (#23847)" (#23953)

This turned out to be mistake in manual submission. Fixed
on JPype1 side.

This reverts commit 3699be49b24ef5a0a8d8de81a149af2c5a7dc206.

(cherry picked from commit 45b5cf004b55f68dcf16dc76e97db893afef76ba)

jedcunningham committed:
Faster grid view (#23951)

(cherry picked from commit c9b21b8026c595878ee4cc934209fc1fc2ca2396)

uranusjr committed:
Disallow calling expand with no arguments (#23463)

(cherry picked from commit 9214018153dd193be6b1147629f73b23d8195cce)

ephraimbuddy committed:
Update version to 2.3.2 and add release notes

ephraimbuddy committed:
Add cascade to `dag_tag` to `dag` foreignkey (#23444)

Bulk delete does not work if the cascade behaviour of a foreignkey
is set on python side(relationship configuration). To allow bulk delete of dags
we need to setup cascade deletion in the DB.

The warning on query.delete at
https://docs.sqlalchemy.org/en/14/orm/session_basics.html#selecting-a-synchronization-strategy
stated that:

The operations do not offer in-Python cascading of relationships - it is assumed that ON UPDATE CASCADE and/or ON DELETE CASCADE is configured for any foreign key references which require it, otherwise the database may emit an integrity violation if foreign key references are being enforced.

Another alternative is avoiding bulk delete of dags but I prefer we support bulk deletes.

This will break offline sql generation for mssql(already broken before now :) ). Also, since there's only one foreign key
in `dag_tag` table, I assume that the foreign key would be named `dag_tag_ibfk_1` in `mysql`. This
avoided having to query the db for the name.

The foreignkey is explicitly named now, would be easy for future upgrades

(cherry picked from commit 8ccff9244a6d1a936d8732721373b967e95ec404)

Taragolis committed:
DagFileProcessorManager: Start a new process group only if current process not a session leader (#23872)

(cherry picked from commit 9216489d9a25f56f7a55d032b0ebfc1bf0bf4a83)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,30 @@
+Doc only changes
+^^^^^^^^^^^^^^^^
+
+- Doc: Add column names for DB Migration Reference (#23853)
```
```suggestion
- Add column names for DB Migration Reference (#23853)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,30 @@
+- Add ``__wrapped__`` property to _TaskDecorator (#23830)
+- Fix UnboundLocalError when sql is empty list in DbApiHook (#23816)
+- Enable clicking on DAG owner in autocomplete dropdown (#23804)
+- Simplify flash message for _airflow_moved tables (#23635)
```
```suggestion
- Simplify flash message for "_airflow_moved" tables (#23635)
```

Maybe?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,30 @@
+- Prevent UI from crashing if grid task instances are null (#23939)
+- Remove redundant register exit signals in ``dag-processor`` command (#23886)
+- Add ``__wrapped__`` property to _TaskDecorator (#23830)
+- Fix UnboundLocalError when sql is empty list in DbApiHook (#23816)
```
```suggestion
- Fix UnboundLocalError when ``sql`` is empty list in DbApiHook (#23816)
```

josh-fell committed:
Introduce `flake8-implicit-str-concat` plugin to static checks (#23873)

(cherry picked from commit 92ddcf4ac6fa452c5056b1f7cad1fca4d5759802)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,36 @@
+- DagFileProcessorManager: Start a new process group only if current process not a session leader (#23872)
+- Mask sensitive values for not-yet-running TIs (#23807)
+- Add cascade to ``dag_tag`` to ``dag`` foreignkey (#23444)
+- Use ``--subdir`` argument value for standalong dag processor. (#23864)
```
```suggestion
- Use ``--subdir`` argument value for standalone dag processor. (#23864)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,36 @@
+- Add missing ``is_mapped`` field to Task response. (#23319)
+- DagFileProcessorManager: Start a new process group only if current process not a session leader (#23872)
+- Mask sensitive values for not-yet-running TIs (#23807)
+- Add cascade to ``dag_tag`` to ``dag`` foreignkey (#23444)
```
```suggestion
- Add cascade to ``dag_tag`` to ``dag`` foreign key (#23444)
```

dskoda1 committed:
[23945] Icons in grid view for different dag types (#23970)

(cherry picked from commit d470a8ef8df152eceee88b95365ff923db7cb2d7)

potiuk committed:
Fix missing shorthand for docker buildx rm -f (#23984)

Latest version of buildx removed -f as shorthand for --force flag.

(cherry picked from commit fc17fbfe9667dbf603f463d028cdffabb2e8f97f)

potiuk committed:
use explicit --mount with types of mounts rather than --volume flags (#23982)

The --volume flag is an old style of specifying mounts used by docker,
the newer and more explicit version is --mount where you have to
specify type, source, destination in the form of key/value pairs.

This is more explicit and avoids some guesswork when volumes are
mounted (for example seems that on WSL2 volume name might be
guessed as path wrongly). The change explicitly specifies which
of the mounts are bind mounts and which are volume mounts.

Another nice side effect of this change is that when source is
missing, docker will not automatically create directories with the
missing name but it will fail. This is nicer because before it
led to creating directories when they were missing (for example
.bash_aliases and similar). This allows us to avoid some cleanups
to account for those files being created - instead we simply
skip those mounts if the file/folder does not exist.

(cherry picked from commit 49367642ef59de06694a80cf35cd6971ebab5eca)

potiuk committed:
Force colors in yarn test output in CI (#23986)

(cherry picked from commit 230a7377490ec9e6bb1c263d53f7f7fa961a3586)

potiuk committed:
Fix breeze failures when there is no buildx installed on Mac (#23988)

If you have no buildx plugin installed on Mac (for example when
you use colima instead of Docker Desktop) the breeze check was
failing - but buildx in fact is not needed to run typical breeze
commands, and breeze already has support for it - it was just
wrongly handled.

(cherry picked from commit 148013b2a88935a2bae5e4b3eb86acb50dd6efc9)

potiuk committed:
Replace generation of docker volumes to be done from python (#23985)

The pre-commit to generate docker volumes in docker compose
file is now written in Python and it also uses the newer "volume:"
syntax to define the volumes mounted in the docker-compose.

(cherry picked from commit 882535a8a2699af7d1d079ecebd8c31aa7fbaba9)

potiuk committed:
Remove pinning for xmltodict (#23992)

We have now moto 3.1.9+ in constraints so we should remove the limit.

Fixes: #23576
(cherry picked from commit eec85d397ef0ecbbe5fd679cf5790adae2ad9c9f)

potiuk committed:
Remove fixing cncf.kubernetes provider when generating constraints (#23994)

When we yanked cncf.kubernetes provider, we pinned 3.1.2
temporarily for provider generation. This removes the pinning as
we are already at 4.0.2 version

(cherry picked from commit 9a9c4009c6480fc2d5c12a38b02225d431a01485)

potiuk committed:
Disable fail-fast on pushing images to docker cache (#24005)

There is an issue with pushing cache to docker registry that
is connected to containerd bug but started to appear more
frequently recently (as evidenced for example by
https://github.community/t/buildx-failed-with-error-cannot-reuse-body-request-must-be-retried/253178
). The issue is still open in containerd:
https://github.com/containerd/containerd/issues/5978.

Until it if fixed, we disable fail-fast on pushing cache
so that even if it happens, we just have to re-run that single
python version that actually failed. Currently there is a much
lower chance of success because all 4 build have to succeed.

(cherry picked from commit ae343fa8e922e369d0d5081894d223133a55b8ef)
